ContactManager2/src/components/ContactForm.jsx

```diff
@@ -0,0 +1,43 @@
+import { InputTextBox } from "./InputTextBox";
+import { contactFieldValidator, contactFormConfig } from "../config/contactFormConfig";
+import { useState } from "react";
+import PropTypes from "prop-types"
+import { v4 as uuidv4 } from "uuid"
+
+export const ContactForm = ({ addContact }) => {
+  const [formData, setFormData] = useState({});
+  const [formError, setFormError] = useState({});
+
+  const updateField = (id, value) => {
+    setFormData(data => ({
+      ...data, [id]: value
+    }))
+  };
+
+  const submitAction = (event) => {
+    event.preventDefault();
+    const errorMsgs = contactFieldValidator.validateAllFields(formData);
+    setFormError(errorMsgs);
+    
+    const noError = Object.keys(errorMsgs).every(key => !Boolean(errorMsgs[key]));
```

Comment:
rename variable name... hasErrorPresent

Reply:
renamed to `hasError` as `hasErrorPresent` doesn't seem like a correct phrase.